src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisCard.tsx

```diff
@@ -0,0 +1,724 @@
+            .pipe(
+              first(),
+              map((v) => v.data.targetNodes[0].recordings.active.data[0] as Recording),
+              tap((recording) => {
+                if (recording == null) {
+                  throw new Error(NO_RECORDINGS_MESSAGE);
+                }
+              })
+            )
+            .subscribe({
+              next: (recording) => {
+                context.reports
+                  .reportJson(recording, target.connectUrl)
+                  .pipe(first())
+                  .subscribe({
+                    next: (report) => {
+                      categorizeEvaluation(report);
+                    },
+                    error: (_) => {
+                      handleStateErrors(FAILED_REPORT_MESSAGE);
+                    },
+                  });
+              },
+              error: (_) => {
+                // try generating report on cached or archived recordings
+                handleEmptyRecordings(target.connectUrl);
+              },
+            })
+        );
+      })
+    );
+  }, [
+    addSubscription,
+    dispatch,
+    context.api,
+    context.target,
+    context.reports,
+    automatedAnalysisAddTargetIntent,
+    setTargetConnectURL,
+    categorizeEvaluation,
+    queryActiveRecordings,
+    handleEmptyRecordings,
+    handleLoading,
+    handleStateErrors,
+  ]);
+
+  const startProfilingRecording = React.useCallback(() => {
+    const config = context.settings.automatedAnalysisRecordingConfig();
+    const attributes = automatedAnalysisConfigToRecordingAttributes(config);
+    addSubscription(
+      context.api.createRecording(attributes).subscribe((resp) => {
+        if (resp && (resp.ok || resp.status === 400)) {
+          // in-case the recording already exists
+          generateReport();
+        } else {
+          handleStateErrors(RECORDING_FAILURE_MESSAGE);
+        }
+      })
+    );
+  }, [
+    addSubscription,
+    context.api,
+    context.settings,
+    context.settings.automatedAnalysisRecordingConfig,
+    generateReport,
+    handleStateErrors,
+  ]);
+
+  const handleErrorView = React.useCallback((): [string, undefined | (() => void)] => {
+    if (errorMessage === NO_RECORDINGS_MESSAGE) {
+      return ['', undefined];
+    } else if (errorMessage === RECORDING_FAILURE_MESSAGE) {
+      return ['Retry starting recording', startProfilingRecording];
+    } else if (errorMessage === FAILED_REPORT_MESSAGE) {
+      return ['Retry loading report', generateReport];
+    } else {
+      // errorMessage === INTERNAL_ERROR_MESSAGE
+      return ['Retry', generateReport];
+    }
+  }, [errorMessage, startProfilingRecording, generateReport]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.target.authFailure().subscribe(() => {
+        handleStateErrors('Authentication failure');
+      })
+    );
+  }, [addSubscription, context.target, handleStateErrors]);
+
+  React.useEffect(() => {
+    generateReport();
+  }, []);
```

Comment:
`generateReport` as deps.

Reply:
Outdated. Look for comments in https://github.com/cryostatio/cryostat-web/pull/618#discussion_r1035649568